### PR TITLE
[fix](nereids)remove literal partition by and order by expression in window function

### PR DESCRIPTION
**Pull request by starocean999**
## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/CheckAfterRewrite.java [hunk 1]
```diff
@@ -70,6 +72,17 @@ private void checkUnexpectedExpression(Plan plan) {
         if (plan.getExpressions().stream().anyMatch(e -> e.anyMatch(SubqueryExpr.class::isInstance))) {
             throw new AnalysisException("Subquery is not allowed in " + plan.getType());
         }
+        if (plan instanceof LogicalWindow) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/CheckAfterRewrite.java:75]: why check here? not in ExtractAndNormalizeWindowExpression?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/CheckAfterRewrite.java [hunk 2]
```diff
@@ -70,6 +72,17 @@ private void checkUnexpectedExpression(Plan plan) {
...
+            if (ExpressionUtils.collect(plan.getExpressions(), WindowExpression.class::isInstance)
+                    .stream()
+                    .anyMatch(expression -> ((WindowExpression) expression).getOrderKeys().stream()
+                                    .map(orderExpression -> orderExpression.getOrderKey().getExpr())
+                                    .anyMatch(Literal.class::isInstance))) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/CheckAfterRewrite.java:80]: why not push them down lick partition by key?

**Comment by starocean999** [comment]
run buildall

**Review by jackwener — DISMISSED** [review]

**Review by morrySnow — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/ExtractAndNormalizeWindowExpression.java [hunk 3]
```diff
@@ -44,7 +46,22 @@ public class ExtractAndNormalizeWindowExpression extends OneRewriteRuleFactory i
...
         return logicalProject().when(project -> containsWindowExpression(project.getProjects())).then(project -> {
-            List<NamedExpression> outputs = project.getProjects();
+            List<NamedExpression> outputs =
+                    ExpressionUtils.rewriteDownShortCircuit(project.getProjects(), output -> {
+                        if (output instanceof WindowExpression) {
+                            // remove literal partition by and order by keys
+                            WindowExpression windowExpression = (WindowExpression) output;
+                            return windowExpression.withPartitionKeysOrderKeys(
+                                    windowExpression.getPartitionKeys().stream()
+                                            .filter(expression -> !(expression instanceof Literal))
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/ExtractAndNormalizeWindowExpression.java:56]: use `expression.isConstant()`

**Review by morrySnow — CHANGES_REQUESTED** [review]

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by englefly — APPROVED** [review]

**Pull request merged**